app.js

```diff
@@ -827,74 +829,94 @@ function sendAccountLinking(recipientId) {
-      encoding: null
-    },
-    (error, response, body) => {
-      if (!error && response.statusCode === 200) {
-        const type = response.headers['content-type'];
-        const prefix = 'data:' + type + ';base64,';
-        // console.log('binary', bl);
-        // const base64 = new Buffer(bl.toString(), 'binary').toString('base64');
-        // const base64 = body.toString('base64');
-        const data = /* prefix + */ body;
-        console.log('base64', body);
-
-        /* This operation detects labels in the supplied image */
-
-        const params = {
-          Image: {
-            Bytes: data
-          },
-          MaxLabels: 123,
-          MinConfidence: 70
-        };
-        rekognition.detectLabels(params, function(err, data) {
-          if (err) {
-            return console.log(err, err.stack); // an error occurred
-          }
-          return callSendAPI({
-            recipient: {
-              id: senderID
-            },
-            message: {
-              text: JSON.stringify(data.Labels),
-              metadata: 'DEVELOPER_DEFINED_METADATA'
-            }
-          });
-          /*
-           data = {
-           Labels: [
-           {
-           Confidence: 99.25072479248047,
-           Name: "People"
-           },
-           {
-           Confidence: 99.25074005126953,
-           Name: "Person"
-           }
-           ]
-           }
-           */
+  let _include_headers = function(body, response, resolveWithFullResponse) {
+    return { headers: response.headers, data: body };
+  };
+  let options = {
+    uri: imageObj.payload.url,
+    encoding: null,
+    method: 'GET',
+    json: true,
+    transform: _include_headers
+  }
+  requestPromise(options)
+    .then(imgResponse => {
+      /* This operation detects labels in the supplied image */
+      const awsPromise = util.promisify(rekognition.detectLabels.bind(rekognition))
+      awsPromise({
+        Image: {
+          Bytes: imgResponse.data
+        },
+        MaxLabels: 123,
+        MinConfidence: 70
+      })
+        .then(awsResponse => {
+          const promises = [];
+
+          awsResponse.Labels.forEach(obj => {
+            promises.push(
+               requestPromise({
+                uri: `https://api.instagram.com/v1/tags/search?q=${obj.Name}&access_token=${process.env.INSTAGRAM_ID}`,
+                encoding: null,
+                method: 'GET',
+                json: true,
+                transform: _include_headers
+              })
+            );
+          });          
+          Promise.all(promises)
+            .then(function(instResponses) {
+              const tags = [];
+              instResponses.forEach((resp)=>{
+                tags.push(...resp.data.data)
+              });
+              
+              // Sorting
+              tags.sort(function(a,b) {return (a.media_count > b.media_count) ? 1 : ((b.media_count > a.media_count) ? -1 : 0);} );               
```

Comment:
arrow function

Reply:
Non gestire lo 0, non serve realmente in questo caso 
`(a,b) => a.media_count > b.media_count ? 1 : -1`